core/src/main/java/hudson/Util.java

```diff
@@ -1310,62 +1294,17 @@ public static File resolveSymlinkToFile(@Nonnull File link) throws InterruptedEx
      */
     @CheckForNull
     public static String resolveSymlink(@Nonnull File link) throws InterruptedException, IOException {
-        try { // Java 7
-            Object path = File.class.getMethod("toPath").invoke(link);
-            return Class.forName("java.nio.file.Files").getMethod("readSymbolicLink", Class.forName("java.nio.file.Path")).invoke(null, path).toString();
-        } catch (NoSuchMethodException x) {
-            // fine, Java 6; fall through
-        } catch (InvocationTargetException x) {
-            Throwable x2 = x.getCause();
-            if (x2 instanceof UnsupportedOperationException) {
-                return null; // no symlinks on this platform
-            }
-            try {
-                if (Class.forName("java.nio.file.NotLinkException").isInstance(x2)) {
-                    return null;
-                }
-            } catch (ClassNotFoundException x3) {
-                assert false : x3; // should be Java 7+ here
-            }
-            if (x2.getClass().getName().equals("java.nio.file.FileSystemException")) {
-                // Thrown ("Incorrect function.") on JDK 7u21 in Windows 2012 when called on a non-symlink, rather than NotLinkException, contrary to documentation. Maybe only when not on NTFS?
-                return null;
-            }
-            if (x2 instanceof IOException) {
-                throw (IOException) x2;
-            }
-            throw (IOException) new IOException(x.toString()).initCause(x);
+        try {
+            Path path =  link.toPath();
+            return Files.readSymbolicLink(path).toString();
+        } catch (UnsupportedOperationException | FileSystemException x) {
+            // no symlinks on this platform (windows?),
+            // or not a link (// Thrown ("Incorrect function.") on JDK 7u21 in Windows 2012 when called on a non-symlink,
+            // rather than NotLinkException, contrary to documentation. Maybe only when not on NTFS?) ?
+            return null;
```

Comment:
need to catch IOException and rethrow to avoid wrapping it in another IOException below.